### PR TITLE
Provide Summary Logging for Payment Dispatch + Cleanup Payment Hashes

### LINKED ISSUE #37: nit: Summary logging for Event Dispatch
Right now we'll [spam](https://github.com/bitcoin-dev-project/sim-ln/blob/main/sim-lib/src/lib.rs#L209) the logs every time we dispatch a payment. Nice for testing, bad for end-users. 

We should bump this down to `trace` and periodically log a summary like: 
`Send 1000 sat over 15 payments from A -> B` 

We can do the same for payments results as they come in.

### LINKED ISSUE #52: nit: Payment hash logging should be hex
```
2023-08-18T05:59:06.981Z INFO [sim_lib] Sending action for PaymentHash([206, 11, 17, 184, 124, 82, 118, 37, 104, 211, 90, 207, 53, 253, 170, 223, 141, 163, 157, 193, 8, 243, 117, 160, 245, 42, 135, 218, 181, 249, 240, 145])
2023-08-18T05:59:06.981Z DEBUG [sim_lib]  Received outcome
2023-08-18T05:59:06.982Z DEBUG [sim_lib] Tracking payment outcome for: PaymentHash([206, 11, 17, 184, 124, 82, 118, 37, 104, 211, 90, 207, 53, 253, 170, 223, 141, 163, 157, 193, 8, 243, 117, 160, 245, 42, 135, 218, 181, 249, 240, 145])
```

### PR DESCRIPTION
**Pull request by carlaKC**
Fixes #52 
Fixes #37

### CODE REVIEW

File: sim-lib/src/lib.rs [hunk 1]
```diff
@@ -418,16 +418,58 @@ async fn consume_simulation_results(
     shutdown: triggered::Trigger,
 ) {
     log::debug!("Simulation results consumer started.");
+    let mut result_logger = PaymentResultLogger::new();
 
     while let Some(resolved_payment) = receiver.recv().await {
-        // TODO - write to CSV.
-        println!("Resolved payment received: {:?}", resolved_payment);
+        result_logger.report_result(&resolved_payment.0, &resolved_payment.1);
+        log::trace!("Resolved payment received: {:?}", resolved_payment);
```
Comment (sr-gi) [inline sim-lib/src/lib.rs:425]: For readability, I think it makes more sense to destructure `resolved_paymeny`

```suggestion
    while let Some((details, result)) = receiver.recv().await {
        result_logger.report_result(&details, &result);
        log::trace!("Resolved payment received: ({:?}, {:?})", details, result);
    }
```

File: sim-lib/src/lib.rs [hunk 2]
```diff
@@ -418,16 +418,58 @@ async fn consume_simulation_results(
...
     }
 
     log::debug!("Simulation results consumer exiting");
     shutdown.trigger();
 }
 
+/// PaymentResultLogger is an aggregate logger that will report on a summary of the payments that have been reported
+/// to it at regular itervals (defined by the log_interval it is created with).
```
Comment (sr-gi) [inline sim-lib/src/lib.rs:433]: nit: intervals

File: sim-lib/src/lib.rs [hunk 3]
```diff
@@ -418,16 +418,58 @@ async fn consume_simulation_results(
...
+            log_interval: 10,
+            ..Default::default()
+        }
+    }
+
+    fn report_result(&mut self, details: &DispatchedPayment, result: &PaymentResult) {
+        match result.failure_reason {
+            Some(_) => self.failed_payment += 1,
+            None => self.success_payment += 1,
+        }
```
Comment (sr-gi) [inline sim-lib/src/lib.rs:456]: I hate empty `Some`

```suggestion
        if result.failure_reason.is_some() {
            self.failed_payment += 1
        } else {
            self.success_payment += 1
        }
```

File: sim-cli/src/main.rs [hunk 4]
```diff
@@ -20,7 +20,7 @@ struct Cli {
 async fn main() -> anyhow::Result<()> {
     SimpleLogger::new()
         .with_level(LevelFilter::Warn)
-        .with_module_level("sim_lib", LevelFilter::Debug)
+        .with_module_level("sim_lib", LevelFilter::Info)
```
Comment (sr-gi) [inline sim-cli/src/main.rs:23]: We should be adding a verbosity flag in a followup

**Review by sr-gi — CHANGES_REQUESTED** [review]
Mainly nits, looks good otherwise

File: sim-lib/src/lib.rs (hunk 3 above)
Comment (carlaKC) [reply sim-lib/src/lib.rs:456]: match is _one_ less LOC so I'm going to leave this one :')

Comment (sr-gi) [reply sim-lib/src/lib.rs:456]: Fair enough

**Review by sr-gi — APPROVED** [review]
LGTM

**Pull request merged**